navigator_helpers/tasks/sample_to_dataset/task_suite.py

```diff
@@ -0,0 +1,887 @@
+        """
+        data_jsonl = sample_dataset.to_json(orient="records", lines=True)
+        data_schema = str(list(sample_dataset.columns))
+
+        dataseed_prompt = DATASEED_REVERSE_ENG_PROMPT_TEMPLATE.format(
+            sampled_dataset_jsonl=data_jsonl,
+            dataset_context_str=dataset_context,
+            sampled_dataset_column_list=data_schema,
+        )
+        if self.config.verbose:
+            print("------------------- Dataseed prompt ------------")
+            print(dataseed_prompt)
+
+        thinking, data_seeds = self.execute_prompt(dataseed_prompt, system_prompt_type)
+
+        return thinking, data_seeds
+
+    def crowdsource_data_seeds(
+        self,
+        sample_dataset: pd.DataFrame,
+        dataset_context: str = "",
+        crowd_size=3,
+        max_num_seeds=3,
+        system_prompt_type="cognition",
+    ) -> dict:
+        """
+        Perform crowd-sourcing by executing the dataseed extraction prompt multiple times,
+        and generating a deduped and ranked list of high-quality data seeds.
+
+        Args:
+            sample_dataset (pd.DataFrame): The sample dataset.
+            dataset_context (str, optional): Dataset context beyond the sample. Defaults to ''
+            crowd_size (int, optional): Number of times to run the user prompt. Defaults to 3.
+            max_num_seeds (int, optional): Maximum number of seeds to return. Defaults to 3.
+            system_prompt_type (str, optional): Type of system prompt to use. Defaults to 'cognition'.
+
+        Returns:
+            dict: A dictionary containing the ranked data seeds.
+
+        Raises:
+            ValueError: If all attempts to extract data seeds fail.
+            RuntimeError: If ranking fails after multiple attempts.
+        """
+
+        # Collect data seeds from crowd_size runs of execute_prompt_w_cognition
+        data_seeds_list = []
+        self.logger.info(f"  |-- Crowd size: {crowd_size}")
+        for i in range(crowd_size):
+            self.logger.info(f"  |-- 🫡 assistant opinion {i+1}")
+            try:
+                _, data_seeds = self.extract_data_seeds_in_one_shot(
+                    sample_dataset, dataset_context, system_prompt_type
+                )
+
+                # Define the model for individual column entries
+                class ExampleDataSeedColumnModel(BaseModel):
+                    column_name: str
+                    description: str
+                    example_values: List[Union[str, int, bool]]
+
+                # Define the model for the entire structure
+                class ExampleDataSeedModel(BaseModel):
+                    columns: List[ExampleDataSeedColumnModel]
+
+                is_valid_data_seeds, validation_result = validate_json_with_pydantic(
+                    ExampleDataSeedModel, data_seeds
+                )
+                if is_valid_data_seeds:
+                    columns = data_seeds.get("columns", None)
```

Comment:
nit: since you already validated these seeds and populated pydantic model with it, this could be just `validation_result.columns`, right?

Though you return a dict, not the pydantic classes, so I guess you would need to serialize these to dict anyways, so this is fine I think.

Reply:
Hmm... also looking at how then these seeds are used, maybe there would be a value in using parsed pydantic objects? Not a big deal though.

Reply:
Arrgh, this one would require way to many changes at the moment, and we may change the overall approach here. Keeping as is for now